Sources/Main.cpp

```diff
@@ -1126,24 +1087,27 @@ void __fastcall TX584Form::CutItemClick(TObject *Sender)
 }
 //---------------------------------------------------------------------------
 
+void TX584Form::CopySelectedItems()
+{
+    TListItem *Item;
+    TItemStates selected = TItemStates() << isSelected;
+
+    Item = CodeListView->Selected;
+    ClipboardSize = 0;
+    for (; Item; Item = CodeListView->GetNextItem(Item, sdBelow, selected)) {
```

Comment:
This code appears multiple times in different functions. I'd suggest to revive the former `GetSelection` function but this time make it return indices of selected items:

```cpp
std::vector<int> TX584Form::GetSelectedItems()
```

Then use it in `CopySelectedItems`, `ClearSelectedItems` and `ClearSelection`.

Reply:
As a side note, please avoid declaring and initializing variables separately:

```cpp
TListItem *Item;
Item = CodeListView->Selected;
```

This is a bad practice. The content of uninitialized primitive values, such as integers, floats, booleans, pointers is undefined, i.e. random garbage. Always declare and initialize such variables at the same time:

```cpp
TListItem *Item = CodeListView->Selected;
```

Also, do not omit initialization statement in `for` loops if possible:

```cpp
for (; Item; Item = CodeListView->GetNextItem(Item, sdBelow, selected)) {
```

The preferred way is the following:

```cpp
for (TListItem *Item = CodeListView->Selected; Item; Item = CodeListView->GetNextItem(Item, sdBelow, selected)) {
```

The loop variable is locally scoped to the loop body and inaccessible outside it.